crates/iota-core/src/authority/shared_object_congestion_tracker.rs

```diff
@@ -220,13 +222,53 @@ pub(crate) struct SharedObjectCongestionTracker {
 }
 
 impl SharedObjectCongestionTracker {
-    pub fn new(mode: PerObjectCongestionControlMode, assign_min_free_execution_slot: bool) -> Self {
+    pub fn new(
+        initial_object_debts: impl IntoIterator<Item = (ObjectID, u64)>,
+        protocol_config: &ProtocolConfig,
+    ) -> Self {
+        let mut object_execution_slots = HashMap::new();
+        for (object_id, debt) in initial_object_debts {
+            let mut slots = ObjectExecutionSlots::new();
+            if debt > 0 {
+                // If there is an initial debt, remove the occupied slot from time 0 to
+                // debt.
+                slots.remove(ExecutionSlot::new(0, debt));
+            }
+            object_execution_slots.insert(object_id, slots);
+        }
```

Comment:
```suggestion
        let object_execution_slots = initial_object_debts
            .into_iter()
            .map(|(object_id, debt)| {
                let mut slots = ObjectExecutionSlots::new();
                if debt > 0 {
                    // If there is an initial debt, remove the occupied slot from time 0 to
                    // debt.
                    slots.remove(ExecutionSlot::new(0, debt));
                }

                (object_id, slots)
            })
            .collect::<HashMap<_, _>>();
```

More importantly, this hash map is computed but used nowhere. Did you mean to use it in line 241 instead of `HashMap::new()`? I guess currently we don't have any debts because `max_congestion_limit_overshoot_per_commit` is not "enabled"? But once `max_congestion_limit_overshoot_per_commit` is set, there is a danger of forgetting to use the computed `object_execution_slots` instead of `HashMap::new()` for initializing object execution slots in the tracker.

Reply:
You're right, this should have been used instead of a new HashMap, thanks for catching that. It was set correctly in the `new_for_test` function, so the problem didn't present itself in the tests.